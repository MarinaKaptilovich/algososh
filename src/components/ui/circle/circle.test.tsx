import renderer from 'react-test-renderer'
import { ElementStates } from '../../../types/element-states'
import { Circle } from './circle'

it('корректный рендер элемента без буквы', () => {
	const tree = renderer.create(<Circle />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('корректный рендер элемента с буквами', () => {
	const tree = renderer.create(<Circle letter='i' />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('корректный рендер элемента с head', () => {
	const tree = renderer.create(<Circle head='i' />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('корректный рендер элемента с react-элементом в head', () => {
	const tree = renderer.create(<Circle head={<></>} />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('корректный рендер элемента с tail', () => {
	const tree = renderer.create(<Circle tail='i' />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('корректный рендер элемента с react-элементом в tail', () => {
	const tree = renderer.create(<Circle tail={<></>} />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('корректный рендер элемента с index', () => {
	const tree = renderer.create(<Circle index={1} />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('корректный рендер элемента с пропом isSmall ===  true', () => {
	const tree = renderer.create(<Circle isSmall={true} />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('корректный рендер элемента default', () => {
	const tree = renderer
		.create(<Circle state={ElementStates.Default} />)
		.toJSON()
	expect(tree).toMatchSnapshot()
})

it('корректный рендер элемента в состоянии changing', () => {
	const tree = renderer
		.create(<Circle state={ElementStates.Changing} />)
		.toJSON()
	expect(tree).toMatchSnapshot()
})

it('корректный рендер элемента в состоянии modified', () => {
	const tree = renderer
		.create(<Circle state={ElementStates.Modified} />)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
