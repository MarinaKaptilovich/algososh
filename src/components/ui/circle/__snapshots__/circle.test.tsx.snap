// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`корректный рендер элемента default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`корректный рендер элемента без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`корректный рендер элемента в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`корректный рендер элемента в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`корректный рендер элемента с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    i
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`корректный рендер элемента с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`корректный рендер элемента с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`корректный рендер элемента с react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`корректный рендер элемента с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    i
  </div>
</div>
`;

exports[`корректный рендер элемента с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      i
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`корректный рендер элемента с пропом isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
